File: gems/decomplex/examples/ruby/temporal-ordering-pressure.rb
# frozen_string_literal: true

class TemporalOrderExample
  def one
    @a = 1
  end

  def two
    @a = 2
    @b = 3
  end

  def three
    @b = 4
  end

  def reader
    @a
  end
end

File: gems/decomplex/examples/php/temporal-ordering-pressure.php
<?php
class TemporalOrderExample {
  public function one() {
    $this->a = 1;
  }

  public function two() {
    $this->a = 2;
    $this->b = 3;
  }

  public function three() {
    $this->b = 4;
  }

  public function reader() {
    return $this->a;
  }
}
